kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/contexts/AsyncThreadProcessorContext.java

```diff
@@ -76,20 +76,14 @@ public class AsyncThreadProcessorContext<KOut, VOut> implements MergedProcessorC
   // time a new event is picked up from the processing queue but before beginning
   // to process it (ie invoking #process on the input record for this event), as
   // part of the preparation for each async process
-  private AsyncEvent currentAsyncEvent;
+  private final AsyncEvent currentAsyncEvent;
```

Comment:
This change makes a lot of sense. I was just blindly following the way the processor contexts work in Streams, which is to have a single context and just mutate its per-record fields as we go through the subtopology. But that mutation has been the direct cause of much grief and added complexity so there's no reason to continue with it here